### PR TITLE
Feature/hockeyapp sync

### PR DESCRIPTION
**Pull request by siepet**
Makes it possible to sync changes based on a diff.

### CODE REVIEW

File: app/models/hockey_app_integration/actions/sync.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      end
+
+      def find_user_id(app, user_email)
+        all_app_users(app).each do |user|
+          return user['id'] if user['email'] == user_email
+        end
+      end
+
+      def sync_add_teams
+        diff[:add_teams].each do |app, teams|
```
Comment (Szeliga) [inline app/models/hockey_app_integration/actions/sync.rb]: You could extract these nested `each` statements to seperate methods `each_team` and `each_user` and use them in the syncing methods. Inside these methods, yield whatever arguments you need

Comment (Szeliga) [inline app/models/hockey_app_integration/actions/sync.rb]: It would also remove the duplication issue from code climate

Comment (siepet) [inline app/models/hockey_app_integration/actions/sync.rb]: Great, sounds good!

Comment (Szeliga) [inline app/models/hockey_app_integration/actions/sync.rb]: You can extract the outer loop as well - `diff[:add_teams].each do |app, teams|` and `diff[:remove_teams].each do |app, teams` - the block signature is the same. You could do `each_team(diff[:remove_teams]) do |app, team_id||`

**Comment by Szeliga** [comment]
LGTM :+1:

**Pull request merged**